.github/workflows/forge-build-check.yml

```diff
@@ -0,0 +1,44 @@
+name: Forge Build Size Check
+
+on:
+  push:
+    paths:
+      - 'packages/contracts-core/**'
+      - '.github/workflows/foundry-build-check.yml'
+  pull_request:
+    paths:
+      - 'packages/contracts-core/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Setup Node JS
+        uses: ./.github/actions/setup-nodejs
+
+      - name: Installing dependencies
+        working-directory: ./packages/contracts-core
+        run: yarn install --immutable
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Run forge build --sizes
+        run: |
+          forge build --sizes
+        working-directory: packages/contracts-core
+        continue-on-error: true # This will ensure the step doesn't fail, and we can check the exit code next
+
+      - name: Check exit code
+        run: |
+          if [ "$?" -eq "1" ]; then
+            echo "forge build --sizes failed!"
+            exit 1
+          fi
```

Comment:
Can we omit this and let the job fail by having a failed `forge build --sizes`? The exit code check doesn't bring any extra granularity here imo.